File: NaoBlocks.Web/Dtos/Robot.cs
using Data = NaoBlocks.Core.Models;

namespace NaoBlocks.Web.Dtos
{
    public class Robot
    {
        public string FriendlyName { get; set; } = string.Empty;

        public bool IsInitialised { get; set; }

        public string MachineName { get; set; } = string.Empty;

        public string? Password { get; set; }

        public static Robot? FromModel(Data.Robot value)
        {
            return value == null ? null : new Robot
            {
                FriendlyName = value.FriendlyName,
                IsInitialised = value.IsInitialised,
                MachineName = value.MachineName
            };
        }
    }
}

File: client/src/app/data/robot.ts
import { RobotLog } from './robot-log';

export class Robot {
    id: string;
    friendlyName: string;
    machineName: string;
    type: string;
    password: string;
    whenAdded: Date;
    isNew: boolean;
    isLoading: boolean;
    isInitialised: boolean;
    logs: RobotLog[];

    constructor(isNew: boolean = false) {
        this.isNew = isNew;
    }
}
